d2bs/kolbot/libs/AutoMule.js

```diff
@@ -435,9 +474,14 @@ MainLoop:
 						(item.classid !== 603 || item.quality !== 7) && // Don't drop Annihilus
 						(item.classid !== 604 || item.quality !== 7) && // Don't drop Hellfire Torch
 						(item.location === 7 || (item.location === 3 && !Storage.Inventory.IsLocked(item, Config.Inventory))) && // Don't drop items in locked slots
-						((!TorchSystem.getFarmers() && !TorchSystem.isFarmer()) || [647, 648, 649].indexOf(item.classid) === -1) && // Don't drop Keys if part of TorchSystem
-						!this.cubingIngredient(item) && !this.runewordIngredient(item) && !this.utilityIngredient(item)) { // Don't drop Runeword/Cubing/CraftingSystem ingredients
-					items.push(copyUnit(item));
+						((!TorchSystem.getFarmers() && !TorchSystem.isFarmer()) || [647, 648, 649].indexOf(item.classid) === -1)) { // Don't drop Keys if part of TorchSystem
+					forced = this.matchItem(item, Config.AutoMule.Force.concat(Config.AutoMule.Trigger));
```

Comment:
If I'm understanding this logic correctly, I think this is what it should be:
```
if (this.matchItem(item, Config.AutoMule.Force.concat(Config.AutoMule.Trigger)) ||
    (this.matchItem(item, Config.AutoMule.Exclude) && (forceCheck || 
    (!this.cubingIngredient(item) && !this.runewordIngredient(item) && !this.utilityIngredient(item))))) {
    items.push(copyUnit(item));
}
```

basically if it is on the force or trigger list, add it to the list, else if its not on the trigger list and is either forceCheck or not a material, then add to items list

Reply:
Actually I can get rid of forceCheck  all together since force list should be prioritized anyways. That was just there to prevent ingredients from triggering mule.